src/internal/permissions.ts

```diff
@@ -36,8 +28,8 @@ export async function getPermissions<T extends PermissionType>({
   aptosConfig: AptosConfig;
   primaryAccountAddress: AccountAddress;
   subAccountPublicKey: Ed25519PublicKey;
-  filter?: T;
-}): Promise<FilteredPermissions<T>> {
+  filter?: new (...a: any) => T;
```

Comment:
Trying something different with the filtering. Eventually we will hit the indexer and we can move away from calling the node directly. 

This takes the class itself, so for example `FungibleAssetPermission` and uses `instanceof` for filtering.